Model/Profile.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace steevanb\SSH2Bundle\Model;
+
+/**
+ * Connection profile
+ */
+class Profile
+{
+    const ADDRESS = 'address';
+    const PORT = 'port';
+    const LOGIN = 'login';
+    const PASSWORD = 'password';
+    const ID_RSA_PUB = 'id-rsa-pub';
+    const ID_RSA_PEM = 'id-rsa-pem';
+    const PASSPHRASE = 'passphrase';
+
+    /** @var string */
+    protected $address;
+
+    /** @var int */
+    protected $port = 22;
+
+    /** @var string */
+    protected $login;
+
+    /** @var string */
+    protected $password;
+
+    /**
+     * @var string
+     */
+    protected $id_rsa_pub;
+
+    /**
+     * @var string
+     */
+    protected $id_rsa_pem;
+
+    /**
+     * @var string
+     */
+    protected $passphrase;
+
+    /**
+     * Profile constructor.
+     * @param array $args
+     */
+    public function __construct(array $args)
+    {
+        foreach ($args as $key => $val) {
+            if ($key === self::ADDRESS ||
+                $key === self::PORT ||
+                $key === self::LOGIN ||
+                $key === self::PASSWORD ||
+                $key === self::PASSPHRASE) {
+
+                $this->{$key} = $val;
+
+            } elseif ($key === self::ID_RSA_PUB ||
+                $key === self::ID_RSA_PEM) {
+
+                $key = str_replace('-', '_', $key);
+                $this->{$key} = $val;
+
+            } else {
+                throw new \InvalidArgumentException(
+                    sprintf('%s is not a valid property in constructor %s',
+                        $key,
+                        self::class
+                    )
+                );
+            }
+        }
+    }
+
+    /**
+     * Return address
+     *
+     * @return string
+     */
+    public function getAddress() : string
```

Comment:
this lib is for PHP 5, so we can't use PHP 7 syntax :(

Reply:
Ok i will remove type